test/Streamly/Test/Data/Unbox.hs

```diff
@@ -37,43 +59,143 @@ data NestedSOP =
+    it "test (Single 0) (Single 1)" $ test (Single 0) (Single 1)
+    it "test (Product2 0 'a') (Product2 1 'b')"
         $ test (Product2 0 'a') (Product2 1 'b')
-    prop "(SOP0) (SOP1 1)" $ test SOP0 (SOP1 1)
-    prop "(SOP0) (SOP2 1 'a')" $ test SOP0 (SOP2 1 'a')
-    prop "(SOP0) (SOP3 1 2 3)" $ test SOP0 (SOP3 1 2 3)
-
-    prop "checkSizeOf Unit" $ checkSizeOf (Proxy :: Proxy Unit) 1
-    prop "checkSizeOf Single" $ checkSizeOf (Proxy :: Proxy Single) 8
-    prop "checkSizeOf Product2" $ checkSizeOf (Proxy :: Proxy Product2) 12
-    prop "checkSizeOf SumOfProducts"
+    it "test (SOP0) (SOP1 1)" $ test SOP0 (SOP1 1)
+    it "test (SOP0) (SOP2 1 'a')" $ test SOP0 (SOP2 1 'a')
+    it "test (SOP0) (SOP3 1 2 3)" $ test SOP0 (SOP3 1 2 3)
+
+    it "checkSizeOf Unit" $ checkSizeOf (Proxy :: Proxy Unit) 1
+    it "checkSizeOf Unit1" $ checkSizeOf (Proxy :: Proxy Unit1) 1
+    it "checkSizeOf Unit2" $ checkSizeOf (Proxy :: Proxy Unit2) 2
+    it "checkSizeOf Unit3" $ checkSizeOf (Proxy :: Proxy Unit3) 17
+    it "checkSizeOf Unit4" $ checkSizeOf (Proxy :: Proxy Unit4) 17
+    it "checkSizeOf UnarySum" $ checkSizeOf (Proxy :: Proxy UnarySum) 1
+    it "checkSizeOf UnarySum2" $ checkSizeOf (Proxy :: Proxy UnarySum2) 2
+    it "checkSizeOf Single" $ checkSizeOf (Proxy :: Proxy Single) 8
+    it "checkSizeOf Product2" $ checkSizeOf (Proxy :: Proxy Product2) 12
+    it "checkSizeOf SumOfProducts"
         $ checkSizeOf (Proxy :: Proxy SumOfProducts) 25
-    prop "checkSizeOf NestedSOP" $ checkSizeOf (Proxy :: Proxy NestedSOP) 26
+    it "checkSizeOf NestedSOP" $ checkSizeOf (Proxy :: Proxy NestedSOP) 26
+
+testBool :: IO ()
+testBool = do
+    sizeOf (Proxy :: Proxy Bool) `shouldBe` genericSizeOf (Proxy :: Proxy Bool)
+    arr <- newUnpinnedBytes (sizeOf (Proxy :: Proxy Bool))
+    let val = True
+    pokeByteIndex 0 arr val
+    genericPeekByteIndex arr 0 `shouldReturn` val
+    genericPokeByteIndex arr 0 val
+    peekByteIndex 0 arr `shouldReturn` val
+
+-- TODO Use shouldReturn in all of the following
+testComplex :: IO ()
+testComplex = do
+    sizeOf (Proxy :: Proxy (Complex Int))
+            `shouldBe` genericSizeOf (Proxy :: Proxy (Complex Int))
+    let val :: (Complex Int) = 5 :+ 3
+    var <- newUnpinnedBytes (sizeOf (Proxy :: Proxy (Complex Int)))
+    pokeByteIndex 0 var val
+    genericPeekByteIndex var 0 `shouldReturn` val
+    genericPokeByteIndex var 0 val
+    peekByteIndex 0 var `shouldReturn` val
+
+-- Ratio does not have a Generic instance by default
+deriving instance Generic (Ratio Int)
+
+testRatio :: IO ()
+testRatio = do
+    sizeOf (Proxy :: Proxy (Ratio Int))
+            `shouldBe` genericSizeOf (Proxy :: Proxy (Ratio Int))
+    let val :: (Ratio Int) = 5 :% 3
+    var <- newUnpinnedBytes (sizeOf (Proxy :: Proxy (Ratio Int)))
+    pokeByteIndex 0 var val
+    genericPeekByteIndex var 0 `shouldReturn` val
+    genericPokeByteIndex var 0 val
+    peekByteIndex 0 var `shouldReturn` val
+
+-- XXX not working for ghc 8.6.5
```

Comment:
Could have put this under a GHC version checking CPP macro.

Reply:
See https://github.com/composewell/streamly/issues/2295